packages/xo-server-sdn-controller/src/index.js

```diff
@@ -0,0 +1,810 @@
+import assert from 'assert'
+import createLogger from '@xen-orchestra/log'
+import NodeOpenssl from 'node-openssl-cert'
+import { EventEmitter } from 'events'
+import { existsSync, readFileSync, writeFile } from 'fs'
+import { filter, find, forOwn, map } from 'lodash'
+import { fromCallback, fromEvent } from 'promise-toolbox'
+
+import { OvsdbClient } from './ovsdb-client'
+
+const log = createLogger('xo:xo-server:sdn-controller')
+
+const PROTOCOL = 'pssl'
+
+const CA_CERT = '/ca-cert.pem'
+const CLIENT_KEY = '/client-key.pem'
+const CLIENT_CERT = '/client-cert.pem'
+
+const SDN_CONTROLLER_CERT = 'sdn-controller-ca.pem'
+
+const NB_DAYS = 9999
+
+exports.configurationSchema = {
+  type: 'object',
+  properties: {
+    'cert-dir': {
+      description: `Full path to a directory where to find: client-cert.pem,
+ client-key.pem and ca-cert.pem to create ssl connections with hosts.
+ If none is provided, the plugin will create its own self-signed certificates.`,
+
+      type: 'string',
+    },
+    'override-certs': {
+      description: `Whether or not to uninstall a SDN controller CA certificate
+ if it already exists in order to replace it with the plugin's ones`,
+
+      type: 'boolean',
+      default: false,
+    },
+  },
+}
+
+// =============================================================================
+
+class SDNController extends EventEmitter {
+  constructor({ xo, getDataDir }) {
+    super()
+
+    this._xo = xo
+
+    this._getDataDir = getDataDir
+
+    this._clientKey = null
+    this._clientCert = null
+    this._caCert = null
+
+    this._poolNetworks = []
+    this._OvsdbClients = []
+    this._newHosts = []
+
+    this._networks = new Map()
+    this._starCenters = new Map()
+
+    this._cleaners = []
+    this._objectsAdded = this._objectsAdded.bind(this)
+    this._objectsUpdated = this._objectsUpdated.bind(this)
+
+    this._overrideCerts = false
+
+    this._unsetApiMethod = null
+  }
+
+  // ---------------------------------------------------------------------------
+
+  async configure(configuration) {
+    this._overrideCerts = configuration['override-certs']
+    let certDirectory = configuration['cert-dir']
+    if (certDirectory == null) {
+      log.debug(`No cert-dir provided, using default self-signed certificates`)
+      certDirectory = await this._getDataDir()
+
+      if (existsSync(certDirectory + CA_CERT) === false) {
```

Comment:
Don't test equality for boolean:
- `cond === true` → `cond`
- `cond === false` → `!cond`